packages/vats/src/provisionPool.js

```diff
@@ -120,24 +115,47 @@ export const start = async (zcf, privateArgs) => {
   // @ts-expect-error vbank purse is close enough for our use.
   const fundPurse = E(poolBank).getPurse(poolBrand);
 
-  // Metrics
-  let totalMintedProvided = AmountMath.makeEmpty(poolBrand);
-  let totalMintedConverted = AmountMath.makeEmpty(poolBrand);
-  /** @type {import('@agoric/inter-protocol/src/contractSupport.js').MetricsPublishKit<MetricsNotification>} */
-  const { metricsPublisher, metricsSubscriber } = makeMetricsPublishKit(
-    privateArgs.storageNode,
-    privateArgs.marshaller,
-  );
-  const updateMetrics = walletsProvisioned => {
-    metricsPublisher.publish(
-      harden({
-        walletsProvisioned,
-        totalMintedProvided,
-        totalMintedConverted,
-      }),
+  const makeMetrics = () => {
+    /** @type {import('@agoric/inter-protocol/src/contractSupport.js').MetricsPublishKit<MetricsNotification>} */
+    const { metricsPublisher, metricsSubscriber } = makeMetricsPublishKit(
+      privateArgs.storageNode,
+      privateArgs.marshaller,
     );
+
+    let walletsProvisioned = 0n;
+    let totalMintedProvided = AmountMath.makeEmpty(poolBrand);
+    let totalMintedConverted = AmountMath.makeEmpty(poolBrand);
+
+    const updateMetrics = () => {
```

Comment:
all these onFoo handlers are updating the (local) metrics. let's use "update" to mean revisions to the truth and "publish" to notify subscribers
```suggestion
    const publishMetrics = () => {
```